ui/components/src/ui-components/inputs/QuantityInput.tsx

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Inputs
+ */
+
+import * as React from "react";
+import classnames from "classnames";
+import { CommonProps } from "@bentley/ui-core";
+import { ParseResults } from "@bentley/ui-abstract";
+import { QuantityStatus } from "@bentley/imodeljs-quantity";
+import { IModelApp, QuantityType } from "@bentley/imodeljs-frontend";
+import { ParsedInput } from "./ParsedInput";
+import { UiComponents } from "../UiComponents";
+
+/** Props for [[QuantityInput]] control
+ * @beta
+ */
+export interface QuantityProps extends CommonProps {
+  /** Initial magnitude in 'persistence' units. See `getUnitByQuantityType` in [QuantityFormatter]($imodeljs-frontend) */
+  initialValue: number;
+  /** Type of quantity being input. */
+  quantityType: QuantityType;
+  /** Function to call in the quantity value is changed. The value returned will be in 'persistence' units. */
+  onQuantityChange: (newQuantityValue: number) => void;
+  /** Set to `true` if value is for display only */
+  readonly?: boolean;
+}
+
+/** Input control that allows users to input a quantity and show the formatted string that represents the value.
+ * @beta
+ */
+export const QuantityInput = React.forwardRef<HTMLInputElement, QuantityProps>(
+  function QuantityInput({ initialValue, quantityType, readonly, className, style, onQuantityChange }, ref) {
+    const [formatterSpec, setFormatterSpec] = React.useState(() => IModelApp.quantityFormatter.findFormatterSpecByQuantityType(quantityType));
+    const [parserSpec, setParserSpec] = React.useState(() => IModelApp.quantityFormatter.findParserSpecByQuantityType(quantityType));
+
+    const formatValue = React.useCallback((value: number) => {

```

Comment:
If `ParsedInput` is adjusted to parse when new formatter is passed in it might make sense to memoize this.

Reply:
Will look at this in the future if needed. I am having issues using useMemo on component that uses React.forwardRef.